Library/Homebrew/api.rb

```diff
@@ -92,21 +92,35 @@ def self.fetch_json_api_file(endpoint, target:)
 
         retry
       end
+
+      if endpoint.end_with?(".jws.json")
+        success, data = verify_and_parse_jws(json_data)
+        unless success
+          target.unlink
+          odie "Failed to verify integrity of #{url} (#{data})."
+        end
+        [data, !skip_download]
+      else
+        [json_data, !skip_download]
+      end
     end
 
-    sig { params(name: String, git_head: T.nilable(String)).returns(String) }
-    def self.fetch_homebrew_cask_source(name, git_head: nil)
+    sig { params(name: String, git_head: T.nilable(String), sha256: T.nilable(String)).returns(String) }
```

Comment:
Wondering if/when we want this to be `nil`?

Reply:
At a minimum: the time period between merging this and the API being updated.